Add Intune Windows quality update profile

FabienTschanz opened the pull request:
#### Pull Request (PR) description
This pull request adds the `IntuneWindowsUpdateForBusinessQualityUpdateProfileWindows10` as a DSC resource to manage the Windows quality updates in Intune. 

**Note**: Currently in draft because create and delete testing is pending due to missing Windows 10/11 Enterprise E3 license on the test tenant. 

#### This Pull Request (PR) fixes the following issues
- Fixes #2659

FabienTschanz committed:
Add Intune Windows quality update profile

ricmestre commented:
It will be a pain also testing this in my automated tests since they run in a dev tenant without the required licensing but anyway...

So I've tested it in my live tenant and there's a backend issue, I created the policy manually, exported it, removed it and re-created it without issues but if I re-run the deployment it will always create a new policy. This is because the cmdlet ```Get-MgBetaDeviceManagementWindowsQualityUpdateProfile``` doesn't support filtering, at least not directly and requires someone to report this to the Graph SDK team, see at the bottom. The workaround is to in ```Get-TargetResource``` if the policy is not found by ```Id``` then retrieve all policies and filter them by ```DisplayName``` locally instead of sending the filter to the API.

```
Get-MgBetaDeviceManagementWindowsQualityUpdateProfile : The query specified in the URI is not valid. Query option 'Filter' is not allowed. To allow it, set the 'AllowedQueryOptions' property on EnableQueryAttribute or QueryValidationSettings.
Status: 400 (BadRequest)
```

FabienTschanz committed:
Replace not applicable filter

FabienTschanz commented:
@ricmestre Should have spotted that one because I replaced the filter in the export method with another way to filter. Updated it now, I will test it tomorrow in my company test tenant.

ricmestre commented on Modules/Microsoft365DSC/DSCResources/MSFT_IntuneWindowsUpdateForBusinessQualityUpdateProfileWindows10/MSFT_IntuneWindowsUpdateForBusinessQualityUpdateProfileWindows10.psm1:
```diff
@@ -94,9 +94,9 @@ function Get-TargetResource
-                $getValue = Get-MgBetaDeviceManagementWindowsQualityUpdateProfile `
-                    -Filter "DisplayName eq '$DisplayName'" `
-                    -ErrorAction SilentlyContinue
+                $getValue = Get-MgBetaDeviceManagementWindowsQualityUpdateProfile | Where-Object -FilterScript {
```
You want to add ```-All``` here to retrieve all profiles first and maybe also keep ```-ErrorAction SilentlyContinue``` otherwise if the cmdlet fails you go straight to the next catch block instead of returning ```$nullResult```.

FabienTschanz replied:
Thanks for being the good eyed watcher. Just updated it, totally makes sense.

FabienTschanz commented:
Checked in my company test tenant, looks good.

FabienTschanz committed:
Merge branch 'Dev' into feat/intune-quality-update

FabienTschanz committed:
Merge branch 'Dev' into feat/intune-quality-update
